hw2/04-background.js

```diff
@@ -1 +1,40 @@
-// Add your code here
+const intervalButton = document.getElementById("intervalButton");
+const numberInput = document.getElementById("numberInput");
+const pageBackground = document.getElementById("pageBackground");
+
+// the colors I wanted to switch between - a color pallete I really liked
+const colors = [
+  "hsla(278, 35%, 78%, 0.3)", // light purple
+  "hsla(337, 100%, 89%, 0.3)", // lighter pipnk
+  "hsla(338, 100%, 84%, 0.3)", // light pink
+  "hsla(208, 97%, 87%, 0.3)", // lighter blue
+  "hsla(209, 100%, 82%, 0.3)", // light blue
+];
+
+let intervalId;
+
+// Derived from: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
+function getRandomInt(max) {
+  return Math.floor(Math.random() * max);
+}
+
+function changeBackground() {
+  // const color = colors[getRandomInt(colors.length)]; // uses the color pallete
+  const color = `hsla(${getRandomInt(360)}, ${getRandomInt(100)}%, ${getRandomInt(100)}%, 0.2)`; // generates any random color with low alpha value
+  pageBackground.style.backgroundColor = color;
+}
+
+function handleClick(event) {
```

Comment:
Here, you can give a condition where it check if the user enters a valid number or not for the interval. If the user doesn't that print a message saying give a correct input or something.

Reply:
Good suggestion! I will add that since users can add negative values or input 0 and it changes the background too quickly. It's probably not too good to have that harsh feature.

Reply:
Thank you for the feedback, fixed this for all my highlighting tasks

Reply:
I agree with Sri Lakshmi's comment here, a validation to check if the user input is correct or would be good .